### ISSUE TITLE
Email Templates in Workflows is Order Dependent (changing order breaks the emails)

### ISSUE THREAD

**Issue by timcortesi** [EscherLabs/Graphene#112]
In workflow email tasks, you can specify a workflow template to use when sending the email.  That seems to work ok, except that if the order of the templates changes, the emails get mangled up as well.  

For example: 
```json
"template": "template_7"
```

This also appears to be true for logic methods as well:
```json
"logic": "method_3",
```

Can we make this more robust, so that adding / reordering / deleting templates and methods doesn't break workflows which require specific indexes for those templates / methods?

**Comment by Cloverstone** [comment]
yeah, this was never intended to stay that way - but will take some work to get around it.
